### ISSUE TITLE
Latest Visual Studio build issues

### ISSUE THREAD

**Issue by drewpts** [OpenXRay/xray-16#279]
Hi guys,

Just had compiled the solution on Windows 10, got issues in couple of projects. Most noticeable I faced is lack of SDL library (SDL.h is not found for example).

Windows 10, VS2017CE, xd_dev branch.

Is it something wrong with my environment or - known issue?

**Comment by eagleivg** [comment]
You doing nuget restore project packages?

**Comment by drewpts** [comment]
No, I just was doing the way documentation suggests :)

**Comment by collectioner** [comment]
Hi,

Please, try the solution provided by @eagleivg 
We'll update documentation as soon as solution of this issue is found.

**Comment by drewpts** [comment]
Hi,

invoked `Update-Package -reinstall` in VS' NuGet shell
SDL problem disappeared, But still see another issues:
![image](https://user-images.githubusercontent.com/10561098/48868029-e679cf80-ede0-11e8-881f-de246d3bfe21.png)

**Comment by drewpts** [comment]
... I am building in Release mode whole solution. Right?

**Comment by collectioner** [comment]
Yes, right. But these errors.. Try first clean whole solution (with removing all bin/obj folders manually) and then build solution again. VS can sometimes show errors like these and sometimes "Clean.." is helpful.

**Comment by drewpts** [comment]
Still remain three issues at the very bottom. Another thousand of issues can disappear on double-click on any of it.

It's only xr_3da not being built.

**Comment by drewpts** [comment]
Sure thing I have 64 bit tools. It feels like my configuration bit broken. Sorry for disturbing!

**Comment by collectioner** [comment]
After short googling, I found this:
https://github.com/simh/simh/issues/479#issuecomment-336233009

I'm not sure, but you can try this variant.

**Comment by drewpts** [comment]
@collectioner 
no, I just realized that wasn't something serious, so I deleted that.

But there is something else:

29>    c:\dev\projects\xray-16\src\xrcore\xrdebug.h(88): fatal error C1001: An internal error has occurred in the compiler.
29>    (compiler file 'd:\agent\_work\3\s\src\vctools\compiler\cxxfe\sl\p1\c\walk.cpp', line 248)
29>     To work around this problem, try simplifying or changing the program near the locations listed above.
29>    Please choose the Technical Support command on the Visual C++
29>     Help menu, or open the Technical Support help file for more information
29>    The command exited with code 2.
29>  Done executing task "CL" -- FAILED.

**Comment by collectioner** [comment]
Following to the official Microsoft documentation, this can be resolved by changing optimization parameters.
Check the link below.
https://msdn.microsoft.com/ru-ru/library/y19zxzb2.aspx

But I think the issue with smth else..
@Xottab-DUTY any suggestions?

**Comment by drewpts** [comment]
There are problems in compilation of BuildStackTrace functions which are declared in xrDebug.h.

Will look into more precisely bit later. Am I only REALLY one who facing this ? Maybe that was because I just updated VS to recent version....

**Comment by collectioner** [comment]
I just updated VS to 15.9.1 version. Will build engine tomorrow, so will see what we'll get.
So, you said only xr_3da has issues?

**Comment by drewpts** [comment]
Yes, xr_3da.
btw, i have v15.9.2

Thanks for your help!

**Comment by collectioner** [comment]
Oh.. My VS still doesn't know about 15.9.2 :D 
Okay, will see tomorrow.

**Comment by PsiKnapp** [comment]
Hey all,

I ran into this issue earlier, I should have posted it. It's a problem with Visual Studio and is broken from 
 version > 15.7.6. I was able to build by getting version 15.7.6, but haven't been able to get it successfully running (lua crashes later on during load).

Here's a link to the bug that I think it is, https://developercommunity.visualstudio.com/content/problem/313306/vs2017-158-internal-compiler-error-msc1cpp-line-15-1.html

It should be fixed in 16.0.

Hope this also works for you.
